samza-sql/src/main/java/org/apache/samza/sql/runner/SamzaSqlApplicationConfig.java

```diff
@@ -214,7 +214,7 @@ private UdfResolver createUdfResolver(Map<String, String> config) {
     Properties props = new Properties();
     props.putAll(domainConfig);
     HashMap<String, String> udfConfig = getDomainProperties(config, CFG_UDF_CONFIG_DOMAIN, false);
-    return new ConfigBasedUdfResolver(props, new MapConfig(udfConfig));
+    return new ReflectionBasedUdfResolver(new MapConfig(udfConfig));
```

Comment:
Sorry, I didn't realize that this is hard-coded here. Could we make this pluggable ? I think calling initializePlugin like above should do it with the udf resolver based arguments.

Reply:
I agree that this is a useful thing to do and discussed it offline. Created a follow-up ticket(SAMZA-2355) for making the UDFResolver pluggable in samza-sql. It will be done in an immediate follow-up patch.